docs/madr/decisions/020-kds-redesign.md

```diff
@@ -0,0 +1,179 @@
+but might have scalability issues for synchronization Zone to Global with large deployments. Zone control-plane needs to send all the Dataplane resources to the global whenever there is a change. That might be a bottleneck that can happen in large deployments. We can use [`Incremental xDS`](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol) which is Envoy's implementation of cache that instead of sending all the resources sends only changes. That could reduce the bandwidth usage.
+
+```proto
+service GlobalKDSService {
+ ...
+ rpc ZoneToGlobalSync(stream envoy.service.discovery.v3.DeltaDiscoveryResponse)
+     returns (stream envoy.service.discovery.v3.DeltaDiscoveryRequest);
+
+ rpc GlobalToZoneSync(stream envoy.service.discovery.v3.DeltaDiscoveryRequest)
+     returns (stream envoy.service.discovery.v3.DeltaDiscoveryResponse);
+
+```
+
+In the first iteration, we can introduce only `ZoneToGlobalSync` with delta xDS, and in the future, we could introduce `GlobalToZoneSync` and also use delta xDS. In the code, I've noticed that we are using our copy of the Snapshot cache and the code is a bit outdated. As a part of that task, we should check if we can change to use `go-control-plane` cache implementation or copy and update our current implementation. 
+
+Maybe not in the first step but we could consider if we can benefit from using different cache implementation `LinearCache` or `MuxCache` for snapshot generation.
+
+### Prioritize operational informations
+
+Certain resources require more frequent synchronization than others. For example, Policies, ZoneIngress, and ZoneEgress must be updated immediately, whereas Insights and Dataplanes objects may be delayed and can be aggregated. To address this, we could consider implementing an option to specify the interval at which the snapshot for specific resources is recalculated. To avoid recalculating changes for all synchronized resources, we need a cache per resource type, which `SimpleCache` should allows us to achieve. In addition, we could set up an RPC service to monitor changes for all resources on separate streams, allowing changes to be processed independently.
+
+### Change the namespace of synchronized resources from `default` to `kuma-system`
+
+Resources synchronized from Zone to Global are created by default in a `default` namespace and we should change it in the new version. It seems that namespace current sync mechanism checks if resources synchronized from upstream are in the store, and in case the namespace is differnet it removes the old ones and create the new one in the new namespace. There should be no cleanup required, we need only to keep current sync mechanism to work with incremental xDS.
+
+Tried:
+- upgrade path
+- rollback
+
+It looks that state returns to the previous one and nothing breaks.
+
+### Handshake
+
+Is it possible for the zone control-plane to be newer than the global control-plane?
```

Comment:
This sections isn't written in a way that would help people reading this document after it's merged. My main question is here after reading this is what is the answer? Can we write this more like:

> It's not possible for the zone CP to be newer than global CP. This is because we can always guarantee that it's the newest [taken from Ilya's answer but why can we guarantee this]. If we were unable to guarantee this, a handshake might work as follows: ...